core/micro_bench/base/array.cpp

```diff
@@ -0,0 +1,252 @@
+    for (auto _ : state) {
+        gko::Array<int> a(x);
+    }
+}
+
+BENCHMARK_REGISTER_F(Array, CopyConstruct)
+    ->RangeMultiplier(8)
+    ->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, MoveConstruct)(benchmark::State &state)
+{
+    auto x = gko::Array<int>{this->exec, state.range(0)};
+    for (auto _ : state) {
+        gko::Array<int> a(std::move(x));
+    }
+}
+
+BENCHMARK_REGISTER_F(Array, MoveConstruct)
+    ->RangeMultiplier(8)
+    ->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, CopyAssign)(benchmark::State &state)
+{
+    auto x1 = gko::Array<int>{this->exec, state.range(0)};
+    auto x2 = gko::Array<int>{this->exec, state.range(0) + 2};
+    for (auto _ : state) {
+        x1 = x2;
+    }
+}
+
+BENCHMARK_REGISTER_F(Array, CopyAssign)
+    ->RangeMultiplier(8)
+    ->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, MoveAssign)(benchmark::State &state)
+{
+    auto x1 = gko::Array<int>{this->exec, state.range(0)};
+    auto x2 = gko::Array<int>{this->exec, state.range(0) + 3};
+    for (auto _ : state) {
+        x1 = std::move(x2);
+    }
+}
+
+BENCHMARK_REGISTER_F(Array, MoveAssign)
+    ->RangeMultiplier(8)
+    ->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, CopyViews)(benchmark::State &state)
+{
+    auto data = new int[state.range(0)];
+    auto data2 = new int[state.range(0)];
+    auto view1 = gko::Array<int>::view(this->exec, state.range(0), data);
+    auto view2 = gko::Array<int>::view(this->exec, state.range(0), data2);
+    for (auto _ : state) {
+        view1 = view2;
+    }
+    delete data, data2;
+}
+
+BENCHMARK_REGISTER_F(Array, CopyViews)->RangeMultiplier(8)->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, MoveViews)(benchmark::State &state)
+{
+    auto data = new int[state.range(0)];
+    auto data2 = new int[state.range(0)];
+    auto view1 = gko::Array<int>::view(this->exec, state.range(0), data);
+    auto view2 = gko::Array<int>::view(this->exec, state.range(0), data2);
+    for (auto _ : state) {
+        view1 = std::move(view2);
+    }
+    delete data, data2;
+}
+
+BENCHMARK_REGISTER_F(Array, MoveViews)->RangeMultiplier(8)->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, TempClone)(benchmark::State &state)
+{
+    auto x = gko::Array<int>{this->exec, state.range(0)};
+    for (auto _ : state) {
+        auto tmp_clone = make_temporary_clone(this->exec, &x);
+    }
+}
+
+BENCHMARK_REGISTER_F(Array, TempClone)->RangeMultiplier(8)->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, Clear)(benchmark::State &state)
+{
+    auto x = gko::Array<int>{this->exec, state.range(0)};
+    for (auto _ : state) {
+        x.clear();
+    }
+}
+
+BENCHMARK_REGISTER_F(Array, Clear)->RangeMultiplier(8)->Range(64, 64 << 12);
+
+
+BENCHMARK_DEFINE_F(Array, ResizeAndReset)(benchmark::State &state)
+{
+    auto x = gko::Array<int>{this->exec, state.range(0)};
+    for (auto _ : state) {
+        x.resize_and_reset(state.range(0) + this->dis(this->gen));
```

Comment:
Can we make this benchmark deterministic instead? Or are you benchmarking `malloc` here?